runtime/plugins/org.jboss.tools.runtime.core/src/org/jboss/tools/runtime/core/extract/ExtractUtility.java

```diff
@@ -41,6 +46,20 @@ public ExtractUtility(File file) {
 			util = new UntarGZUtility(file);
 		}
 	}
+	public ExtractUtility(File file, int format) {
+		this.file = file;
+		switch(format) {
+			case FORMAT_ZIP:
+				util = new UnzipUtility(file);
+				break;
+			case FORMAT_TAR:
+				util = new UntarUtility(file);
+				break;
+			case FORMAT_TGZ:
+				util = new UntarGZUtility(file);
+				break;
+		}
```

Comment:
The .extract() method will return an IStatus.  If it can't discover how to extract, it will return an error status. If it attempts to extract using the wrong format (ie unzip a tgz file) the underlying extraction will throw a CoreException, which ExtractUtility catches and wraps with an error status. 

So I think the rest of existing api there is sufficient for now.